js/src/nav-menu.js

```diff
@@ -4,101 +4,190 @@
+						cb.checked = true;
+					}
+
+					label.prepend( cb ); // phpcs:ignore WordPressVIPMinimum.JS.HTMLExecutingFunctions.prepend
+				} );
+			} );
+		},
+
+		/**
+		 * Creates and returns a `<input type=hidden"/>` element.
+		 *
+		 * @param {String}         id     An identifier for this input. It will be part of the final `id` attribute.
+		 * @param {Integerg}       itemId The ID of the menu element (post ID).
+		 * @param {String|Integer} value  The input's value.
+		 * @returns {HTMLElement}
+		 */
+		createHiddenInput: ( id, itemId, value ) => {
+			return pllNavMenu.printMetabox.createElement( 'input', {
+				type:  'hidden',
+				id:    `edit-menu-item-pll-${ id }-${ itemId }`,
+				name:  `menu-item-pll-${ id }[${ itemId }]`,
+				value: value
+			} );
+		},
+
+		/**
+		 * Creates and returns an element.
+		 *
+		 * @param {String} type Element's type.
+		 * @param {Object} atts Element's attributes.
+		 * @returns {HTMLElement}
+		 */
+		createElement: ( type, atts ) => {
+			const el = document.createElement( type );
+			for ( const [ key, value ] of Object.entries( atts ) ) {
+				el.setAttribute( key, value );
 			}
-		);
+			return el;
+		}
+	},
+
+	ensureContent: {
+		/**
+		 * Attaches an event to `#menu-to-edit` to disallow unchecking both `show_names` and `show_flags`.
+		 */
+		attachEvent: () => {
+			document.getElementById( 'menu-to-edit' ).addEventListener( 'change', ( event ) => {
+				if ( ! event.target.id || event.target.checked ) {
+					// Now checked, nothing to do.
+					return;
+				}
+
+				const matches = event.target.id.match( new RegExp( /^edit-menu-item-show_(names|flags)-(\d+)$/ ) );
+
+				if ( ! matches ) {
+					// Not the checkbox we want.
+					return;
+				}
+
+				// Check the other checkbox.
+				const other = 'names' === matches[1] ? 'flags' : 'names';
+				document.getElementById( `edit-menu-item-show_${ other }-${ matches[2] }` ).checked = true;
+			} );
+		}
+	},
+
+	showHideRows: {
+		/**
+		 * Attaches an event to `#menu-to-edit` to show or hide the `hide_current` checkbox when `dropdown` is checked.
+		 */
+		attachEvent: () => {
+			document.getElementById( 'menu-to-edit' ).addEventListener( 'change', ( event ) => {
+				if ( ! event.target.id ) {
+					// Not the checkbox we want.
+					return;
+				}
+
+				const matches = event.target.id.match( new RegExp( /^edit-menu-item-dropdown-(\d+)$/ ) );
+
+				if ( ! matches ) {
+					// Not the checkbox we want.
+					return;
+				}
+
+				const hideCb = document.getElementById( `edit-menu-item-hide_current-${ matches[1] }` );
+
+				if ( ! hideCb ) {
+					// Should not happen.
+					return;
+				}
+
+				if ( event.target.checked ) {
+					// Hide and uncheck.
+					hideCb.closest( '.description' ).classList.add( 'hidden' );
+					hideCb.checked = false;
+				} else {
+					// Show.
+					hideCb.closest( '.description' ).classList.remove( 'hidden' );
+				}
+			} );
```

Comment:
```suggestion
				const description = hideCb.closest( '.description' );
                description.classList.toggle( 'hidden', event.target.checked );
                hideCb.checked = ! event.target.checked;
```
I suggest a shorter and more idiomatic manner.

Reply:
Done in [b57afd2](https://github.com/polylang/polylang/pull/1583/commits/b57afd22b834b1364d74a2a06fddca44bd54d899).

However, I didn't do `hideCb.checked = ! event.target.checked;` because this would mean that the checkbox is checked when we show it: if it was not checked before I hide it (by checking the other checkbox), I wouldn't want it to be checked when it is shown back.

Also, I triggered a `change` event in [c688e89](https://github.com/polylang/polylang/pull/1583/commits/c688e89c6b26b3446aac6739709b3ab2b87e8091).

Reply:
No, but it's better to fire it.